### PR TITLE
Add 'npm run watch' support for client and server

### PR DESCRIPTION
**Pull request by bqqbarbhg**
This PR allows you to run `npm run watch` on the server side and automatically rebuild the client bundles on changes.

Also included is fix for a few bugs:
- Server was always using the default configuration file
- Webpack output directory was wrong

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "lint": "eslint .",
     "build": "cross-env NODE_ENV=production webpack -p --mode production --config webpack-config/webpack.production.config.js",
     "build:dev": "cross-env NODE_ENV=development webpack --mode development --config webpack-config/webpack.development.config.js",
+    "watch": "cross-env NODE_ENV=development webpack --mode development --config webpack-config/webpack.development.config.js --watch",
```
Comment (oamaok) [inline client/package.json:12]: could be technically just `"watch": "npm run build:dev -- --watch"`, and the `--mode` should be in the configuration file as you have files per environment to begin with.

Comment (bqqbarbhg) [reply client/package.json:12]: Yep, but I thought having a shorthand would be nicer. @quvide didn't like my `--mode` spam either but I don't trust Webpack so I want to set the mode any way I can: `NODE_ENV`, `-p`, `--mode`, and a separate config file with one more `mode` directive.

Comment (bqqbarbhg) [reply client/package.json:12]: Oh I misunderstood, you can recursively use `npm` scripts in other ones, that's way better!

Comment (bqqbarbhg) [reply client/package.json:12]: Actually turns out that this adds like 0.5-1s to the startup time. Minimal one time code quality improvement is not worth the small performance loss and extra console spam every time these commands are run.

**Comment by HenrikHelenius** [comment]
Nothing to comment, push others to get approved

**Review by joonalillfors — APPROVED** [review]
Looks good to me

**Pull request merged**